packages/fileeditor-extension/src/commands.ts

```diff
@@ -872,6 +876,16 @@ export namespace Commands {
       isEnabled: () => Boolean(isEnabled() && tracker.currentWidget?.content),
       label: trans.__('Select All')
     });
+
+    /**
+     * Override Esc that causes documentsearch:end, doing nothing
+     */
+    commands.addCommand(CommandIDs.overrideEsc, {
+      execute: () => {
+        return;
+      },
+      label: trans.__('Override Esc')
+    });
```

Comment:
Instead of adding this command we probably should modify how `documentsearch:end` behaves, e.g. by modifying selector.

Reply:
I'm not sure we want to change the documentsearch:end... Its selector is jp-mod-searchable, which includes the container of the whole file editor content page (as it should be, I think). 
That is, if I remove the overrideEsc, then pressing esc anywhere else on the sheet (when neither completer or search widget is opened) will go back to the documentsearch:end warning. The warning comes up because indeed no search widget is enabled. One could get rid of the warning  :D 
I can remove the completer-active part of the selector, that way Esc will always pass through the completer:dismiss-file, even when no completer has been invoked (but maintain the end command if a search widget is opened).
Override Esc or always use completer (or the Escape binding from some other package), I can't come up with a third option...